app/controllers/feedback_controller.rb

```diff
@@ -29,30 +29,31 @@ def show
   private
 
   def provider_email
-    if params[:signed_out_provider_id].present?
-      signed_out_provider_email
-    else
-      signed_in_provider_email
-    end
+    # citizen side won't have access to current_provider but can get provider_email via current_application_id
+    provider&.email || current_provider&.email
   end
 
-  def signed_out_provider_email
-    provider_id = params[:signed_out_provider_id]
-    Provider.find(provider_id).email
+  def provider
+    LegalAidApplication.find_by_id(application_id)&.provider
   end
 
-  def signed_in_provider_email
-    provider_struct = session['warden.user.provider.key']
-    return nil unless provider_struct
+  def application_id # rubocop:disable Metrics/AbcSize
+    return params['application_id'] if params['application_id']
+    return session['current_application_id'] if source == :citizen
+
+    application_id_from_page_history || nil
+  end
 
-    provider_id = provider_struct.first&.first
-    return nil unless provider_id
+  def application_id_from_page_history
+    previous_page = session['page_history'][-2]
```

Comment:
We're not storing page history in the session any more. You'll need to get it from redis instead. (It looks like you're on an old branch, you'll need to rebase to pick up the changes).

We store a `page_history_id` in the session now, which can be used by the `PageHistoryService` to get the history, so this code would now be something like:
```
page_history_service = PageHistoryService.new(page_history_id: session[:page_history_id])
page_history = page_history_service.read
previous_page = page_history[-2]
```
(you might be able to refactor that a bit)